kmir/src/kmir/kdist/mir-semantics/alloc.k

```diff
@@ -12,15 +15,28 @@ module ALLOC
   imports INT
 
 syntax BinderForExistentialTraitRef ::= binderForExistentialTraitRef(value: ExistentialTraitRef, boundVars: BoundVariableKindList)
-syntax MaybeBinderForExistentialTraitRef ::= someBinderForExistentialTraitRef(BinderForExistentialTraitRef)
-                                           | "noBinderForExistentialTraitRef"
-syntax GlobalAlloc ::= globalAllocFunction(Instance)
-                     | globalAllocVTable(Ty, MaybeBinderForExistentialTraitRef)
-                     | globalAllocStatic(StaticDef) [symbol(globalAllocStatic)]
-                     | globalAllocMemory(Allocation) [symbol(globalAllocMemory)]
-
-syntax GlobalAllocEntry ::= globalAllocEntry(MIRInt, GlobalAlloc) [symbol(globalAllocEntry)]
-syntax GlobalAllocsMap ::= List {GlobalAllocEntry, ""} [symbol(globalAllocsMap), terminator-symbol(.globalAllocsMap)]
+                                          [group(mir---value--bound-vars)]
+
+syntax MaybeBinderForExistentialTraitRef ::= someBinderForExistentialTraitRef(BinderForExistentialTraitRef) [group(mir-option)]
+                                           | "noBinderForExistentialTraitRef"                               [group(mir-option)]
+
+syntax GlobalAllocKind ::= globalAllocFunction(Instance)
+                             [group(mir-enum), symbol(GlobalAllocKind::Function)]
+                         | globalAllocVTable(Ty, MaybeBinderForExistentialTraitRef)
+                             [group(mir-enum), symbol(GlobalAllocKind::VTable)]
+                         | Static(StaticDef)
+                             [group(mir-enum), symbol(GlobalAllocKind::Static)]
+                         | Memory(Allocation)
+                             [group(mir-enum), symbol(GlobalAllocKind::Memory)]
+
+syntax GlobalAlloc ::= globalAllocEntry(MIRInt, GlobalAllocKind)
+         [symbol(globalAllocEntry), group(mir)]
+
+syntax GlobalAllocs ::= List {GlobalAlloc, ""}
+         [symbol(globalAllocsMap), terminator-symbol(.globalAllocsMap), group(mir-list)]
+
+////////////////////////////// unused for parsing?
+
```

Comment:
This was used to hold the list stored in the allocs field of the json input. I don't know if this has been removed in more recent versions.
If it is needed, the list symbols should be updated: `GlobalAllocs::append`, `GlobalAllocs::empty`

Reply:
Indeed, we have to align those symbols with the rest, thanks.

Reply:
As before, `GlobalAllocs` holds the data from the `allocs` field in the json, it was just renamed. The `unused` comment was referring to the syntax below, I have removed it (if this is dead code we clean it up later).